### DISCUSSION TITLE
Trame on remote Jupyter notebook has no interaction

### DISCUSSION THREAD

**Discussion by zetwal** [pyvista/pyvista#6466] [Q&A] [answered]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

```
I cannot get the client or server backend to work. The end result is I get no interaction.
```

### Description

Hello

I am trying to use Trame on a remote jupyter notebook and I get no interactivity, the only thing that works is the static backend.

My code is as follows, the error for each option is in the comment:
```python
import sys
sys.path.append('venv/lib/python3.11/site-packages')

import pyvista as pv

sphere = pv.Sphere()
sphere.plot(jupyter_backend='trame')

# long example
plotter = pv.Plotter(notebook=True)
plotter.add_mesh(sphere)
#plotter.show(jupyter_backend='server') #404: Not Found
#plotter.show(jupyter_backend='client') #404: Not Found
plotter.show(jupyter_backend='static') # static image shown
```

If I do this, I get different error messages but again only static works:
```python
import pyvista as pv

sphere = pv.Sphere()
sphere.plot(jupyter_backend='trame')

pv.global_theme.trame.server_proxy_enabled = True
pv.global_theme.trame.server_proxy_prefix = '<remote-server-name>:'


# long example
plotter = pv.Plotter(notebook=True)
plotter.add_mesh(sphere)
plotter.show(jupyter_backend='server') #The address wasn’t understood
#plotter.show(jupyter_backend='client') # white screen
#plotter.show(jupyter_backend='static') # static image shown
```

As I mentioned, I'm doing remote connection. So, I initially login to the front-end nodes of the cluster, and then request and allocation. Once on the allocation, I do:
export PYVISTA_TRAME_SERVER_PROXY_PREFIX='/proxy/'
```bash
ipython kernel install --name "venv" --user
jupyter-notebook --no-browser --port=8871 --ip=0.0.0.0 &
```

On my client, I do the following:
`ssh -N -f -L 8871:<node-allocated>:8871 usenamer@remote-server`

and then open a browser as: http://localhost:8871

There is no x-server or GPUs on these nodes

Here are my relevant modules:
```bash
ipykernel                6.23.1
ipython                  8.14.0
ipython-genutils         0.2.0
ipywidgets               8.1.3
isoduration              20.11.0
jedi                     0.18.2
Jinja2                   3.1.2
jsonpointer              2.0
jsonschema               4.17.3
jupyter_core             5.3.0
jupyter-events           0.6.3
jupyter_server           2.6.0
jupyter_server_proxy     4.3.0
jupyter_server_terminals 0.4.4
jupyterlab-pygments      0.2.2
jupyterlab_widgets       3.0.11
 .
 .
pyvista                  0.44.1
 .
 .
 .
trame                    3.6.3
trame-client             3.2.1
trame-server             3.0.3
trame-vtk                2.8.9
trame-vuetify            2.6.2
typing_extensions        4.12.2
tzdata                   2024.1
uri-template             1.2.0
urllib3                  2.2.2
vtk_osmesa               9.3.1
wcwidth                  0.2.7
```

How do I get interaction?

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Tue Aug 06 09:09:41 2024 MDT

                  OS : Linux (Red Hat Enterprise Linux 8.10)
              CPU(s) : 72
             Machine : x86_64
        Architecture : 64bit
                 RAM : 125.6 GiB
         Environment : Jupyter
         GPU Details : error
    MathText Support : False

  Python 3.11.6 (main, Dec  7 2023, 10:25:07) [GCC 9.4.0]

             pyvista : 0.44.1
                 vtk : 9.3.1
               numpy : 1.26.1
          matplotlib : 3.8.1
              scooby : 0.10.0
               pooch : 1.8.2
              pillow : 10.0.0
             IPython : 8.14.0
          ipywidgets : 8.1.3
               trame : 3.6.3
        trame_client : 3.2.1
        trame_server : 3.0.3
           trame_vtk : 2.8.9
       trame_vuetify : 2.6.2
jupyter_server_proxy : 4.3.0
        nest_asyncio : 1.5.6
--------------------------------------------------------------------------------
```

**Comment by tkoyama010** [comment]
Thank you for posting. If you don't have x-server on your environment, Please try vtk-osmesa.

https://discourse.vtk.org/t/status-update-vtk-python-wheels/11212

```shell
pip install pyvista
pip uninstall vtk -y
pip install --extra-index-url https://wheels.vtk.org vtk-osmesa
```

**Reply by zetwal** [reply]
I already did that; I only have vtk-osmesa and not vtk

**Reply by tkoyama010** [reply, marked as answer]
OK. The other thing that I want to point is:
1. `sys.path.append('venv/lib/python3.11/site-packages')` is not recommended to use. Please use venv as follow to avoid environment trouble.

```bash
python -m venv venv
source venv/bin/activate
pip install pyvista[all]
pip uninstall vtk
pip install --extra-index-url https://wheels.vtk.org vtk-osmesa
pip install  jupyterlab
jupyter lab
```
```python
import pyvista as pv

m = pv.Sphere()
m.plot()
```

2. If 1. is not working. Please try "html" backend. This backend will be limited, but will provide interactive plotting.
 
```python
import pyvista as pv

pyvista.set_jupyter_backend("html")

m = pv.Sphere()
m.plot()
```

**Reply by zetwal** [reply]
It's fixed.

I created a new virtual environment, clean install, and I am not using "sys.path.append('venv/lib/python3.11/site-packages')" anymore. I have interaction.
